frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/CodeListTableEditor.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import type { CodeListItem } from '@studio/components';
+import type { Option } from 'app-shared/types/Option';
+import { useTranslation } from 'react-i18next';
+import { StudioCodeListEditor, StudioModal } from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from '../../../../../hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from '../../../../../hooks/queries/useOptionListsQuery';
+import {
+  convertCodeListItemListToOptionsList,
+  convertOptionsListToCodeListItemList,
+} from './utils/conversionUtils';
+import { useCodeListEditorTexts } from './hooks/useCodeListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './CodeListTableEditor.module.css';
+
+type CodeListTableEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function CodeListTableEditor({ component }: CodeListTableEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { doReloadPreview } = usePreviewContext();
+  const { data: optionsListMap, isFetching } = useOptionListsQuery(org, app);
+  const { mutate: uploadOptionList } = useUpdateOptionListMutation(org, app, {
+    hideDefaultError: true,
+  });
+  const [codeListItemList, setCodeListItemList] = useState<CodeListItem[]>([]);
```

Comment:
Not sure I understand the usage completely as I have not looked much into the code connected to either the old or the new codelists, but should not the initial state be based on the potential optionList that is connected to the component?

Reply:
very true, i could set the default value here and not have it as a useEffect... i'm a little embarassed i didn't think about that. I'll change it

Reply:
Okey, so this is option-editing that is independent of what option list that is connected to the component? Because also the predefined option lists can be set on the component as a reference to optionListId. But I might misunderstand what you mean. Maybe we can clear it ut in a short meeting on Monday? 😄